packages/server/src/service-manager.ts

```diff
@@ -33,7 +35,32 @@ export class ServiceWorkerManager implements IServiceWorkerManager {
     return this._ready.promise;
   }
 
+  private unregisterOldServiceWorkers = () => {
+    // Check if we have an installed version. If we do, compare it to the current version
+    // and unregister all service workers if they are different.
+    const installedVersion = localStorage.getItem('jupyterlite-version');
```

Comment:
Not sure how this approach would work if multiple JupyterLite deployments are hosted under the same host? Could there be an issue accessing the same local storage, if for example there is a JupyterLite site on 0.2.2 and another one on 0.2.3?

Reply:
An example is a GitHub organization having multiple repo with a JupyterLite deployment on GitHub Pages.

Reply:
Yeah good point. this can be a problem if they share the domain (thus localstorage) like you say.

If site A has 0.2.2 and is loaded, then site B with 0.2.3 is loaded will override version stored in local storage to 0.2.3. Now if site A is updated to new version (0.2.3) it will wrongly _not_ unregister its old version (0.2.2).

What we can do is switch out the localstorage key to include e.g. the full path to the service worker which thus will be unique across the two deployments even if they are on the same domain?